dali/kernels/reduce/reduce_drop_dims_test.cc

```diff
@@ -0,0 +1,196 @@
+// Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+#include "dali/kernels/reduce/reduce_drop_dims.h"
+
+namespace dali {
+namespace kernels {
+namespace reduce_impl {
+
+TEST(DropDims, Simplify) {
+  DropDims dd;
+  unsigned in_mask = 0b10011, out_mask = 0xffffffffu;
+  int64_t in_shape[5] = { 2, 3, 4, 5, 6 };
+  int64_t out_shape[5];
+  int d = dd.simplify(out_shape, out_mask, in_shape, in_mask);
+  EXPECT_EQ(d, 3);
+  EXPECT_EQ(out_mask, 0b101);
+  EXPECT_EQ(out_shape[0], 2*3);
+  EXPECT_EQ(out_shape[1], 4*5);
+  EXPECT_EQ(out_shape[2], 6);
+
+  in_mask = 0b00110;
+  out_mask = 0xffffffffu;
+  d = dd.simplify(out_shape, out_mask, in_shape, in_mask);
+  EXPECT_EQ(d, 3);
+  EXPECT_EQ(out_mask, 0b010);
+  EXPECT_EQ(out_shape[0], 2);
+  EXPECT_EQ(out_shape[1], 3*4);
+  EXPECT_EQ(out_shape[2], 5*6);
+
+  in_mask = 0;
+  out_mask = 0xffffffffu;
+  d = dd.simplify(out_shape, out_mask, in_shape, in_mask);
+  EXPECT_EQ(d, 1);
+  EXPECT_EQ(out_mask, 0);
+  EXPECT_EQ(out_shape[0], 2*3*4*5*6);
+}
+
+TEST(DropDims, NoOp) {
+  DropDims dd;
```

Comment:
With the default constructor used isn't the content of `div`, `mul` and `mod` undefined?

Reply:
If `start == kMaxDim * 2` the arrays are never accessed.